modules/core/src/main/java/org/apache/ignite/internal/processors/query/GridQueryProcessor.java

```diff
@@ -2618,7 +2619,15 @@ else if (bin)
 
         if (binaryVal) {
             int typeId = ctx.cacheObjects().typeId(val);
-
+            //In the case of using a simple name mapper, the typeId may be different, and we
```

Comment:
New test passes without this changes.
Can you, please, assist me with the test that fail without changes?